File: Python/safadao.py
#
# author: @vpcsilva
#

def somatorio(mes):
    return sum( range( 1, mes + 1 ) )

def wesley_safadao(dia, mes, ano):
    safadeza = somatorio(mes) + (ano/100) * (50-dia)
    anjo = 100.0 - safadeza

    print anjo, "%% ANJO, mas o", safadeza, "%% eh VAGABUNDO"

print "Insira sua data de nascimento:"
dia = int(raw_input("Dia (dd): "))
mes = int(raw_input("Mes (mm): "))
ano = float(raw_input("Ano (AA): "))
wesley_safadao(dia, mes, ano)

File: R/safadao.R
#
# Awesome Safadão
# vpcsilva/AwesomeSafadao
# author: @ronaldosvieira
#

somatorio <- function(mes) {
  soma = 0
  for (i in 1:mes) {
    soma <- soma + i
  }
  
  soma
}

wesley_safadao <- function(dia, mes, ano) {
  safadeza <- somatorio(mes) + (ano / 100) * (50 - dia)
  print(safadeza)
  anjo <- 100 - safadeza
  
  print(paste("Você é ", safadeza, "% safado(a) e ", anjo, "% anjo!", sep=""))
}

print("Insira sua data de nascimento\n")

dia <- as.integer(readline("Dia (0-31):"))
mes <- as.integer(readline("Mês (0-12):"))
ano <- as.integer(readline("Ano (00-99):"))

wesley_safadao(dia, mes, ano)
